src/lib/structures/errors.ml

```diff
@@ -252,6 +257,18 @@ let report_mode_error fmt = function
   | Forbidden_command s ->
     fprintf fmt "Command %s" s
 
+let report_model_error ppf = function
```

Comment:
I believe raising these indicates an internal error, in which case we should say so in the error message (otherwise the user will not understand why we are trying to substitute these identifiers, since they didn't ask).

Reply:
It's not necessary an internal error as the user could apply `subst` on a model while using the library. I prefix the error message with `Model error: ` to clarify where the exception could be raised.